src/packages/solid/components/Metadata/Rating.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { Component } from 'solid-js';
+import { Text, Show } from '@lightningjs/solid';
+import type { IntrinsicNodeProps } from '@lightningjs/solid';
+import Icon from '../Icon/Icon';
+import type { IconProps } from '../Icon/Icon';
+import styles from './Rating.styles';
+
+export interface RatingProps extends IntrinsicNodeProps {
+  /**
+   * path to the rating's icon
+   */
+  src: NonNullable<IconProps['src']>;
+  /**
+   * Text or number to display. Numbers from 0 to 100 will display as percentages.
+   */
+  title: string;
+}
+
+const Rating: Component<RatingProps> = (props: RatingProps) => {
+  const formatTitle = (title: string | number) => {
+    if ((typeof title !== 'string' && typeof title !== 'number') || !String(title).trim().length) {
+      return;
+    }
+
+    let formatted = title;
+    if (!Number.isNaN(title) && Number(title) >= 0 && Number(title) <= 100) {
+      formatted += '%';
+    }
+    return formatted;
+  };
+  return (
+    <>
+      <Show when={props.src}>
+        <Icon
+          src={props.src}
+          marginRight={formatTitle(props.title) ? styles.iconMarginRight : props.marginRight}
```

Comment:
Here you just want to check for props.title? Are you worried about invalid titles?

Reply:
https://docs.solidjs.com/references/api-reference/basic-reactivity/createMemo - formattedTitle() would then be a value you could just use. Haven't done this yet - but assume this is the correct path forward.

Reply:
Invalid or just an empty string. The margin between the icon and text of a rating is different from the margin between 2 Rating components. So if there isn't any text after the icon, the margin between ratings needs to be used instead of the margin between icon|text.

That makes sense to memoize though. I was getting warnings from Stencil when trying to get the the value outside the jsx to reference in the multiple places.